### PR TITLE
iptables: fix error type-casting

### PR DESCRIPTION
**Pull request by sanbornm**
#35

### CODE REVIEW

File: iptables/iptables.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ func (e *Error) ExitStatus() int {
 }
 
 func (e *Error) Error() string {
-	return fmt.Sprintf("exit status %v: %v", e.ExitStatus(), e.msg)
+	return fmt.Sprintf("iptables exit status %v: %v", e.ExitStatus(), e.msg)
```
Comment (lucab) [inline iptables/iptables.go]: Minor: we can also run ip6tables here. Please just leave this out of your PR, we have https://github.com/coreos/go-iptables/pull/36/files in progress covering that.

File: iptables/iptables.go [hunk 2]
```diff
@@ -248,7 +248,12 @@ func (ipt *IPTables) runWithOutput(args []string, stdout io.Writer) error {
 	}
 
 	if err := cmd.Run(); err != nil {
-		return &Error{*(err.(*exec.ExitError)), stderr.String()}
+		switch err.(type) {
+		default:
+			return err
+		case *exec.ExitError:
```
Comment (lucab) [inline iptables/iptables.go]: Do you mind switching the order of cases? It's idiomatic to have the default case as last.

File: iptables/iptables.go [hunk 3]
```diff
@@ -248,7 +248,12 @@ func (ipt *IPTables) runWithOutput(args []string, stdout io.Writer) error {
...
+			return &Error{*(err.(*exec.ExitError)), stderr.String()}
```
Comment (bboreham) [inline iptables/iptables.go]: if you put a variable in on line 251, e.g. `switch e := err.(type)` then you don't need to repeat the cast here.

**Comment by lucab** [comment]
Bump. @sanbornm may you address review comments and rebase to master?

File: iptables/iptables.go (hunk 1 above)
Comment (squeed) [inline iptables/iptables.go]: indeed, PR36 was merged and improves this line

**Comment by lucab** [comment]
@sanbornm it looks like something went wrong with your latest update and this PR now contains a bunch of spurious commits. I think you should just squash all your changes in a single commit on top of current master.

**Comment by heartlock** [comment]
@sanbornm Any update?

**Comment by lucab** [comment]
Closing in favor of https://github.com/coreos/go-iptables/pull/39 to unblock a bug-fix.

**Pull request closed without merging**